testing/execution_engine_integration/src/nethermind.rs

```diff
@@ -0,0 +1,117 @@
+use crate::build_utils;
+use crate::execution_engine::GenericExecutionEngine;
+use crate::genesis_json::nethermind_genesis_json;
+use std::path::{Path, PathBuf};
+use std::process::{Child, Command, Output};
+use std::{env, fs::File};
+use tempfile::TempDir;
+use unused_port::unused_tcp_port;
+
+const NETHERMIND_BRANCH: &str = "kiln";
+const NETHERMIND_REPO_URL: &str = "https://github.com/NethermindEth/nethermind";
+
+fn build_result(repo_dir: &Path) -> Output {
+    Command::new("dotnet")
+        .arg("build")
+        .arg("src/Nethermind/Nethermind.sln")
+        .arg("-c")
+        .arg("Release")
+        .current_dir(repo_dir)
+        .output()
+        .expect("failed to make nethermind")
+}
+
+pub fn build(execution_clients_dir: &Path) {
+    let repo_dir = execution_clients_dir.join("nethermind");
+
+    if !repo_dir.exists() {
+        // Clone the repo
+        assert!(build_utils::clone_repo(
+            execution_clients_dir,
+            NETHERMIND_REPO_URL
+        ));
+    }
+
+    // Checkout the correct branch
+    assert!(build_utils::checkout_branch(&repo_dir, NETHERMIND_BRANCH));
+
+    // Update the branch
+    assert!(build_utils::update_branch(&repo_dir, NETHERMIND_BRANCH));
+
+    // Build nethermind
+    build_utils::check_command_output(build_result(&repo_dir), "dotnet build failed");
+
+    // Build nethermind a second time to enable Merge-related features.
+    // Not sure why this is necessary.
+    build_utils::check_command_output(build_result(&repo_dir), "dotnet build failed");
```

Comment:
Without this 2nd build, Nethermind errors with:
```
{"jsonrpc":"2.0","error":{"code":-32601,"message":"Method engine_exchangeTransitionConfigurationV1 is not supported"},"id":1}
```
so it seems like the `engine` modules isn't being built the first time.

There may be some trickiness with Git not properly checking out the branch, but I'm not sure why is works for Geth.
Happy to hear suggestions.

Reply:
Checked with the Nethermind folks. The issue seems to be that the first build doesn't copy the `plugins` folder in bin. 
The engine api doesn't get enabled without the `merge` plugin.

Building twice seems to build the `plugins` folder which enables the engine api. So I think this is probably the cleanest way to get the tests working until it's fixed on Nethermind.